cmd/waku/server/rest/relay.go

```diff
@@ -44,11 +35,8 @@ func NewRelayService(node *node.WakuNode, m *chi.Mux, cacheCapacity int, log *za
 		node:          node,
 		log:           log.Named("relay"),
 		cacheCapacity: cacheCapacity,
```

Comment:
Are we not going to use a cache for the messages?  if that's the case, then we can remove the `cacheCapacity` parameter from this function and the attribute from `RelayService` struct. I have no preference between one vs the other, just wondering about it because this would be differnt from nwaku.

The difference between having the cache instead of relying on the subscription was that you could always have the last N more recent messages in the cache (since it worked kinda like a ring buffer), compared to using the subscription in which you'll always have the oldest 1000 messages, (iirc)

Reply:
There is another PR #869  where i had updated to use this cacheCapacity to pass it to subscription itself. Since now subscription itself acts like a cache, I don't see the need to maintain this second cache at REST level.

Reply:
But the difference still remains. The subscription will fill itself up with the first `cacheCapacity` messages and then stop receiving messages, while other mechanism you'd always have the last `cacheCapacity` mssages received.

Dunno if it might cause problems like being disconnected due to being a `subscriber too slow`, since perhaps the REST method will not be called frequently enough

Reply:
Aha, then maybe we can enhance subscription to take an option of over-riding older msgs in case of full?
Because if consumer is slow, it would make sense to take some action at subscription level itself. This way REST API or a regular API user gets same behaviour. 
I am also thinking we should align the behaviour between REST API and programmatic (golang or c-bindings) API.

Reply:
> maybe we can enhance subscription to take an option of over-riding older msgs in case of full

This sounds great! I'll open an issue for this and approve the PR then!